src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_ARRAY_SIZE = 10;
+    private int size = 0;
+    private int index;
+    private Object[] array = new Object[START_ARRAY_SIZE];
+
     @Override
     public void add(T value) {
-
+        if (size >= array.length) {
+            array = Arrays.copyOf(array, (int) (array.length * 1.5));
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index: "
+                    + index + ", Size: " + size);
+        } else {
```

Comment:
we don't need `else` clause, throwing means exiting the method, unwrap the logic inside of it